src/day8/re_frame/trace.cljs

```diff
@@ -144,31 +144,42 @@
     (fn [trace]
       (< (:query query) (:duration trace)))))
 
-(defn render-traces [showing-traces trace-detail-expansions]
+(defn render-traces [showing-traces filter-items trace-detail-expansions]
   (doall
     (for [{:keys [op-type id operation tags duration] :as trace} showing-traces]
-      (let [padding   {:padding "0px 5px 0px 5px"}
-            row-style (merge padding {:border-top (case op-type :event "1px solid lightgrey" nil)})
+      (let [padding          {:padding "0px 5px 0px 5px"}
+            row-style        (merge padding {:border-top (case op-type :event "1px solid lightgrey" nil)})
             show-row? (get-in @trace-detail-expansions [:overrides id]
                         (:show-all? @trace-detail-expansions))
+            op-name          (if (vector? operation)
+                               (second operation)
+                               operation)
+            save-query       (fn [filter-for-string]
+                               (swap! filter-items conj {:id (random-uuid)
```

Comment:
Can you refactor the other save-query in render-trace-panel to use this one to save?